### ISSUE TITLE
Not working - 923+, DSM 7.2-64570 Update 1, Samsung 980 Pro

### ISSUE THREAD

**Issue by kennedybaird** [007revad/Synology_enable_M2_volume#99]
Followed all instructions, rebooted, not detecting the drive at all.

When I ran the `hdd_db.sh` script, it said no M.2 drives found.

**Comment by 007revad** [comment]
Shut down the Synology then check the M.2 drive is seated correctly. When you boot the Synology you should see the M.2 in storage manager shown as a cache drive.

**Comment by kennedybaird** [comment]
Nothing showing unfortunately. Drive is definitely seated correctly. Also, took it out, placed in another machine and checked it was working + formatted again.

![image](https://github.com/007revad/Synology_enable_M2_volume/assets/20593811/cc4605f5-4f11-423f-bd19-60ecd9c4f672)

Should I run through the steps again..?

**Comment by 007revad** [comment]
> Should I run through the steps again..?

It wouldn't hurt. A few people have needed to run this script (or the Synology_HDD_db script) a few times before it worked.

You could also try this other script as well: https://github.com/007revad/Synology_HDD_db

**Comment by MischaM26** [comment]
just finished installation of same M.2, same DSM version on a 723+. Works flawlessly.
Thanks a lot Dave!

**Comment by kennedybaird** [comment]
@007revad sorry to pester again, but still having no luck with either script:

![image](https://github.com/007revad/Synology_enable_M2_volume/assets/20593811/555403c9-5eda-4c2d-a066-b3ff5dbdb568)


Running multiple times I get the same results, no change in logs, no change in the drives available through UX

**Comment by 007revad** [comment]
Sorry I closed this issue. I forgot there were 2 different people commenting.

Did your Samsung 980 Pro should up in storage manager as a cache drive before you ran either script?

Run both scripts with their --restore option and answer y to let the syno_enable_m2_volume script reboot the DS923+
```
sudo -i /volume1/data/scripts/syno_hdd_db.sh --restore
sudo -i /volume1/data/scripts/syno_enable_m2_volume.sh --restore
```
Then check that storage manager shows the Samsung 980 Pro as a cache drive in the Built-In M.2 Slot.
![image](https://github.com/007revad/Synology_enable_M2_volume/assets/39733752/dcbafb5d-26e3-4f49-958b-f2449fa8615a)

I just had a thought. Maybe if you only have 1 NVMe drive it needs to be in slot 1. See the red circled raised 1 and 2 in this image: https://imgur.com/jJPDBUg 

Do the following command list any NVMe drives?

`ls /dev/nvme*`

`sudo -i nvme list`

**Comment by kennedybaird** [comment]
syno_hdd_db,sh hung after the ds923+_host_v7.db, so I had to kill the process.

Yes the HDD is stored in slot 1

![image](https://github.com/007revad/Synology_enable_M2_volume/assets/20593811/10d5340e-8c71-422b-855b-9a5f4e66bcae)


--

After rebooting, I tried re-running and i see an extra log underneath "M.2 volume support already enabled"
![image](https://github.com/007revad/Synology_enable_M2_volume/assets/20593811/4cb20e31-c473-430d-abfc-8de2cab888b0).

--

Now the Storage Manager doesn't seem to exist

![image](https://github.com/007revad/Synology_enable_M2_volume/assets/20593811/f0d986b8-3991-4be5-891e-b1d058a26cc4)

**Comment by kennedybaird** [comment]
Also only seeing a restart option up here, no shutdown available now:

![image](https://github.com/007revad/Synology_enable_M2_volume/assets/20593811/e51a483c-b0b1-4aa5-9668-34df5c684f5d)

**Comment by 007revad** [comment]
DSM definitely is not seeing your Samsung 980 Pro. 

Can you disable the scheduled tasks for both scripts. Next shut down the DS923+ and move the Samsung 980 Pro to slot 2. Then boot up and check if storage manager now shows your NVMe drive.

**Comment by 007revad** [comment]
OP not responding

**Comment by kennedybaird** [comment]
Sorry for not replying, had a bit on.

Ok, so this was really starting to bug me. There was weird behaviour, my Storage Manager kept disappearing every time I ran the restore option on this script.

I decided to do a full reset of my device. The 980 Pro is still not detected at all. I pulled an old Intel SSD out of another laptop, and that is working.

I ran the script again on the completely new setup, with the Intel SSD showing in drives and I am able to create a storage pool with the Intel drive.

Going to run down to electronics store and get an enclosure to try some things with this 980 Pro, and get my drives for the storage pool.
